src/system-health/health_checker/sysmonitor.py

```diff
@@ -0,0 +1,634 @@
+
+#Checks the unit status and updates the system status
+def check_unit_status(event, st_db, config_db):
+    global SYSTEM_STATE
+    global SYSTEM_ALLSRV_STATE
+    global dnsrvs_name_list 
+    global allsrvs_status
+    global allsrvs_dict
+
+    astate="DOWN"
+    cstate="DOWN"
+    msg=""
+
+    #check for all status
+    full_srv_list=get_all_service_list(config_db)
+    #print("full srv list:{}".format(full_srv_list))
+    if event in full_srv_list:
+        ustate = get_unit_status(event, st_db, config_db)
+        if ustate == "OK" and SYSTEM_ALLSRV_STATE == "UP":
+            astate = "UP"
+        elif ustate == "OK" and SYSTEM_ALLSRV_STATE == "DOWN":
+            if event in dnsrvs_name_list:
+                #print("remove {} from dnsrvs_name_list".format(event))
+                dnsrvs_name_list.remove(event)
+                #need to check if need to set cstate to UP if this was the only down service before, which became UP now.
+                if len(dnsrvs_name_list) == 0:
+                    astate = "UP"
+                else:
+                    astate = "DOWN"
+        else:
+            if event not in dnsrvs_name_list:
+                dnsrvs_name_list.append(event)
+            astate = "DOWN"
+
+        if astate == "DOWN":
+            msg = "System is not ready - one or more services are not up"
+            allsrvs_status = "DOWN"
+        elif astate == "UP":
+            msg = "System is ready"
+            allsrvs_status = "UP"
+        #logger.log_info("all - event:{}  ustate:{} astate:{} dnsrvs:{}".format(event,ustate,astate,dnsrvs_name_list))
+
+        if SYSTEM_ALLSRV_STATE != astate:
+            SYSTEM_ALLSRV_STATE=astate
+            logger.log_notice(msg)
+            print_console_message(msg)
+            post_system_status(astate, st_db)
+    else:
+        #if received event is not in current full service list but exists in global dictionary & list,
+        #then it should be removed from dictionary & list
+        srv_name,last_name = event.split('.')
+        if allsrvs_dict.__contains__(srv_name):
+            allsrvs_dict.pop(srv_name)
+
+        #also remove from dnsrvslist
+        if event in dnsrvs_name_list:
+            dnsrvs_name_list.remove(event)
+
+def execute_command(command, keep_std_err=False, safe=False):
+    proc = subprocess.Popen(command, stdout=subprocess.PIPE,
+                            #Using shell=True is unsafe in general and should be avoided
+                            shell=(not safe),
+                            universal_newlines=True,
+                            stderr=(subprocess.STDOUT if not keep_std_err else subprocess.PIPE))
+    try:
+        stdout, error = proc.communicate(timeout=30)
+        proc.wait()
+        return stdout, error
+
+    except subprocess.TimeoutExpired:
+        logger.log_error("TimeoutExpired for {}".format(command))
+
+
+class MonitorStateDbTask(object):
```

Comment:
Could you please consider inheriting from ProcessTaskBase in src/sonic-py-common/sonic_py_common/task_base.py? BTW, what is the purpose to use a process instead of a thread?

Reply:
Used task_base.py as per suggestion.
Process is better in handling queue events send and receive as I saw some issues with threads.